Moises.botarro/better handling wildcards data scrubber

moisesbotarro opened the pull request:
@DataDog/burrito 
This PR prevent the user from using '*' as a wildcard to match all the process arguments. 

It also fixes the bugs:

1. Matching the middle of an argument key that contains '-<sensitive_word>', as in 
1-password --config=12345 => 1-password ********

2. Matching multiple arguments as an argument key if they don't contain the next expected character. 
Example:
\*password would match all the bolded arguments on 
java **kafka.kafka dont_have letter_after_m -password** 12345  
as a key

conorbranagan commented on config/data_scrubber.go:
```diff
@@ -49,6 +49,11 @@ func CompileStringsToRegex(words []string) []*regexp.Regexp {
 		}
 
+		if word == "*" {
+			log.Warnf("data scrubber: %s not compiled. The sensitive word '*' is not supported", word)
```
let's say `ignoring wildcard-only ("*") sensitive word as it is not supported` instead. Upon further review I'd change the message for the other warning in a similar way as "skipped" is more meaningful than "not compiled" to a user, IMO.

moisesbotarro commented on config/data_scrubber_test.go:
```diff
@@ -41,14 +43,14 @@ func setupDataScrubberWildCard(t *testing.T) *DataScrubber {
 
 	expectedPatterns := make([]string, 0, len(defaultSensitiveWords))
 	for _, word := range append(defaultSensitiveWords) {
```
The setup functions already use the public constructor on 
```
scrubber := NewDefaultDataScrubber()
scrubber.AddCustomSensitiveWords(wildcards)
```

I created the regex expressions on the test just to verify if these two methods above created the expected regex patterns on the DataScrubber object for the default and custom words. 
The idea was that the private setup functions test the internals of dataScrubber and the other public Test functions test the interface.

Do you think that it's better to get rid off the regex pattern asserts since they test the internals?

moisesbotarro commented:
Looking at the agent on the staging, I realized that the argument 
**‑Dopscenter.ssl.keyStorePassword=opscenter** on https://dd.datad0g.com/process?columns=host,process,user,cpu,memory,start&options=normalizeCPU,showArguments&sort=memory,DESC&text=password 
is not hiden although using \*password\* as a sensitve word.

moisesbotarro commented:
In the case where we use `*path*` as a sensitive word and we have a process with cmdline

/usr/bin/java -Xms18397m  -cp $ES_CLASS**PATH**:$ES_HOME/lib/\*:$ES_HOME/lib/sigar/*:/usr/local/elasticsearch-1.7.6/lib/elasticsearch-1.7.6.jar:/usr/local/elasticsearch-1.7.6/lib/*:/usr/local/elasticsearch-1.7.6/lib/sigar/\* org.elasticsearch.bootstrap.Elasticsearch

it will hide the exe name since the value of cp is matched as an arg key.  

So, `$` was added as a forbidden character in the pattern to match an arg key. I've also added `/` as a forbidden character.

I'd also added `:` as a possible delimiter along ` ` and `=`.
